.github/workflow-gen/Program.cs

```diff
@@ -180,9 +183,19 @@ public static void StepSetupDotNet(this Job job)
             .Name("Setup .NET")
             .ActionsSetupDotNet("3e891b0cb619bf60e2c25674b222b8940e2c1c25", ["6.0.x", "8.0.x", "9.0.x"]); // v4.1.0
 
+    /// <summary>
+    /// Only run this for a main build
+    /// </summary>
     public static Step IfRefMain(this Step step) 
         => step.If("github.ref == 'refs/heads/main'");
 
+    /// <summary>
+    /// Only run this if the build is triggered on a branch IN the same repo
+    /// this means it's from a trusted contributor.
+    /// </summary>
+    public static Step OnlyIfOnBranch(this Step step)
```

Comment:
`IfEventIsPush` ?

Reply:
Sure, I thought it would be clearer to indicate the intent, rather than the methodology, but this is more in line with the rest of the code. No strong preference, so I changed it.